python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -0,0 +1,1150 @@
+    _DefaultName = "gbdesAstrometricFit"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.makeSubtask("sourceSelector")
+        self.makeSubtask("referenceSelector")
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # We override runQuantum to set up the refObjLoaders
+        inputs = butlerQC.get(inputRefs)
+
+        instrumentName = butlerQC.quantum.dataId['instrument']
+
+        sampleRefCat = inputs['referenceCatalog'][0].get()
+        refEpoch = sampleRefCat[0]['epoch']
+
+        refConfig = LoadReferenceObjectsConfig()
+        refConfig.anyFilterMapsToThis = "phot_g_mean"
+        refConfig.requireProperMotion = True
+        refObjectLoader = ReferenceObjectLoader(dataIds=[ref.datasetRef.dataId
+                                                         for ref in inputRefs.referenceCatalog],
+                                                refCats=inputs.pop('referenceCatalog'),
+                                                config=refConfig,
+                                                log=self.log)
+
+        output = self.run(**inputs, instrumentName=instrumentName, refEpoch=refEpoch,
+                          refObjectLoader=refObjectLoader)
+
+        for outputRef in outputRefs.outputWcs:
+            visit = outputRef.dataId['visit']
+            butlerQC.put(output.outputWCSs[visit], outputRef)
+        butlerQC.put(output.outputCatalog, outputRefs.outputCatalog)
+
+    def run(self, inputCatalogRefs, inputVisitSummary, instrumentName=None, refEpoch=None,
+            refObjectLoader=None):
+        """Run the WCS fit for a given set of visits
+
+        Parameters
+        ----------
+        inputCatalogRefs : `list`
+            List of `DeferredDatasetHandle`s pointing to visit-level source
+            tables.
+        inputVisitSummary : `list` of `lsst.afw.table.ExposureCatalog`
+            List of catalogs with per-detector summary information.
+        instrumentName : `str`, optional
+            Name of the instrument used.
+        refEpoch : `float`
+            Epoch of the reference objects in MJD.
+        refObjectLoader : instance of
+            `lsst.meas.algorithms.loadReferenceObjects.ReferenceObjectLoader`
+            Referencef object loader instance.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            ``outputWCSs`` : `list` of `lsst.afw.table.ExposureCatalog`
+                List of exposure catalogs (one per visit) with the WCS for each
+                detector set by the new fitted WCS.
+            ``fitModel`` : `wcsfit.WCSFit`
+                Model-fitting object with final model parameters.
+            ``outputCatalog`` : `pyarrow.Table`
+                Catalog with fit residuals of all sources used.
+        """
+        self.log.info("Gathering instrument, exposure, and field info")
+        # Set up an instrument object
+        instrument = wcsfit.Instrument(instrumentName)
```

Comment:
If instrument is optional, what happens if you call this with `None`?

Reply:
Addressed in docs.